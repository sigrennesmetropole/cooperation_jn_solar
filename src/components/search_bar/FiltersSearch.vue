<script setup lang="ts">
import CheckBox from '@/components/simulation/CheckBox.vue'
import deleteSquare from '@/assets/icons/delete-square.svg'
import { ref } from 'vue'
import type { Ref } from 'vue'

const props = defineProps<{
  isDisplayFilters: boolean
  filters: {
    name: string
    active: boolean
  }[]
}>()

const buttonIsFocus: Ref<boolean> = ref(false)
</script>

<template>
  <div
    class="flex flex-col rounded-lg mt-1 shadow-lg bg-white w-[402px] px-6 py-6"
    v-if="props.isDisplayFilters"
  >
    <div class="flex flex-row items-start justify-between">
      <span class="font-dm-sans font-bold text-2xl mb-6">
        Param√®tres de recherche
      </span>
      <button
        class="flex items-center justify-between rounded p-1 cursor-pointer"
        :class="{
          'bg-neutral-600': buttonIsFocus,
        }"
        @click="$emit('closeFilters')"
        @keyup.enter="$emit('closeFilters')"
        @focus="buttonIsFocus = true"
        @blur="buttonIsFocus = false"
      >
        <img class="w-4 h-4" :src="deleteSquare" />
      </button>
    </div>

    <div class="flex flex-col gap-1 ml-4">
      <div
        class="flex flex-row items-end font-dm-sans font-normal text-sm color-black"
        v-for="(filter, index) in props.filters"
        :key="index"
      >
        <CheckBox
          :isOnError="false"
          :isChecked="filter.active"
          @checkBoxChange="
            $emit('checkBoxChange', {
              value: $event,
              filter: filter,
              index: index,
            })
          "
        >
          <template v-slot:text> </template>
        </CheckBox>
        <label :for="filter.name">{{ filter.name }}</label>
      </div>
    </div>
  </div>
</template>
