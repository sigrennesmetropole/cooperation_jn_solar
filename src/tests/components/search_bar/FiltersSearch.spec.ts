import FiltersSearch from '@/components/search_bar/FiltersSearch.vue'
import { mount, VueWrapper } from '@vue/test-utils'
import { beforeEach, describe, afterEach, expect, it } from 'vitest'

describe('FiltersSearch.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(FiltersSearch, {
      propsData: {
        isDisplayFilters: true,
        filters: [
          {
            name: 'Communes',
            active: false,
          },
          {
            name: 'Adresses',
            active: true,
          },
          {
            name: 'Voies',
            active: false,
          },
          {
            name: 'Organismes',
            active: true,
          },
        ],
      },
    })
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('renders the component', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays the filters', () => {
    const filters = wrapper.findAll('label')
    expect(filters).toHaveLength(4)
    expect(filters[0].text()).toBe('Communes')
    expect(filters[1].text()).toBe('Adresses')
    expect(filters[2].text()).toBe('Voies')
    expect(filters[3].text()).toBe('Organismes')
  })

  it('emits closeFilters event when close button is clicked', async () => {
    const closeButton = wrapper.find('button')
    await closeButton.trigger('click')

    expect(wrapper.emitted()).toHaveProperty('closeFilters')
    expect(wrapper.emitted().closeFilters).toHaveLength(1)
  })

  it('emits closeFilters event when pressing Enter key on close button', async () => {
    const closeButton = wrapper.find('button')
    await closeButton.trigger('keyup.enter')

    expect(wrapper.emitted()).toHaveProperty('closeFilters')
    expect(wrapper.emitted().closeFilters).toHaveLength(1)
  })

  it('sets buttonIsFocus to true when button receives focus', async () => {
    const closeButton = wrapper.find('button')
    await closeButton.trigger('focus')
    // @ts-ignore
    expect(wrapper.vm.buttonIsFocus).toBe(true)
  })

  it('sets buttonIsFocus to false when button loses focus', async () => {
    const closeButton = wrapper.find('button')
    await closeButton.trigger('focus')
    // @ts-ignore
    expect(wrapper.vm.buttonIsFocus).toBe(true)

    await closeButton.trigger('blur')
    // @ts-ignore
    expect(wrapper.vm.buttonIsFocus).toBe(false)
  })
})
