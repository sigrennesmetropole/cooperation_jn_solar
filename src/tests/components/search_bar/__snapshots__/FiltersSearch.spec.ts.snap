// Vitest Snapshot v1

exports[`FiltersSearch.vue > renders correctly 1`] = `
"<div class=\\"flex flex-col rounded-lg mt-1 shadow-lg bg-white w-[402px] px-6 py-6\\">
  <div class=\\"flex flex-row items-start justify-between\\"><span class=\\"font-dm-sans font-bold text-2xl mb-6\\"> ParamÃ¨tres de recherche </span><button class=\\"flex items-center justify-between rounded p-1 cursor-pointer\\"><img class=\\"w-4 h-4\\" src=\\"/src/assets/icons/delete-square.svg\\"></button></div>
  <div class=\\"flex flex-col gap-1 ml-4\\">
    <div class=\\"flex flex-row items-end font-dm-sans font-normal text-sm color-black\\">
      <div class=\\"flex p-0 gap-3 items-top\\">
        <div class=\\"shadow-sm w-5 h-5 rounded border flex justify-center items-center mt-[3px] border-slate-400\\"><input type=\\"checkbox\\" class=\\"opacity-0 absolute\\"><svg class=\\"w-3 h-3 pointer-events-none stroke-slate-900 stroke-2\\" viewBox=\\"0 0 14 14\\" style=\\"display: none;\\">
            <path d=\\"M0.5 8.5498L3.23 12.0598C3.32212 12.1795 3.44016 12.2768 3.57525 12.3443C3.71034 12.4119 3.85898 12.4479 4.01 12.4498C4.15859 12.4515 4.3057 12.4201 4.44063 12.3578C4.57555 12.2956 4.6949 12.204 4.79 12.0898L13.5 1.5498\\" fill=\\"none\\"></path>
          </svg></div>
        <div class=\\"flex w-fit\\"></div>
      </div><label for=\\"Communes\\">Communes</label>
    </div>
    <div class=\\"flex flex-row items-end font-dm-sans font-normal text-sm color-black\\">
      <div class=\\"flex p-0 gap-3 items-top\\">
        <div class=\\"shadow-sm w-5 h-5 rounded border flex justify-center items-center mt-[3px] border-slate-900\\"><input type=\\"checkbox\\" class=\\"opacity-0 absolute\\"><svg class=\\"w-3 h-3 pointer-events-none stroke-slate-900 stroke-2\\" viewBox=\\"0 0 14 14\\" style=\\"\\">
            <path d=\\"M0.5 8.5498L3.23 12.0598C3.32212 12.1795 3.44016 12.2768 3.57525 12.3443C3.71034 12.4119 3.85898 12.4479 4.01 12.4498C4.15859 12.4515 4.3057 12.4201 4.44063 12.3578C4.57555 12.2956 4.6949 12.204 4.79 12.0898L13.5 1.5498\\" fill=\\"none\\"></path>
          </svg></div>
        <div class=\\"flex w-fit\\"></div>
      </div><label for=\\"Adresses\\">Adresses</label>
    </div>
    <div class=\\"flex flex-row items-end font-dm-sans font-normal text-sm color-black\\">
      <div class=\\"flex p-0 gap-3 items-top\\">
        <div class=\\"shadow-sm w-5 h-5 rounded border flex justify-center items-center mt-[3px] border-slate-400\\"><input type=\\"checkbox\\" class=\\"opacity-0 absolute\\"><svg class=\\"w-3 h-3 pointer-events-none stroke-slate-900 stroke-2\\" viewBox=\\"0 0 14 14\\" style=\\"display: none;\\">
            <path d=\\"M0.5 8.5498L3.23 12.0598C3.32212 12.1795 3.44016 12.2768 3.57525 12.3443C3.71034 12.4119 3.85898 12.4479 4.01 12.4498C4.15859 12.4515 4.3057 12.4201 4.44063 12.3578C4.57555 12.2956 4.6949 12.204 4.79 12.0898L13.5 1.5498\\" fill=\\"none\\"></path>
          </svg></div>
        <div class=\\"flex w-fit\\"></div>
      </div><label for=\\"Voies\\">Voies</label>
    </div>
    <div class=\\"flex flex-row items-end font-dm-sans font-normal text-sm color-black\\">
      <div class=\\"flex p-0 gap-3 items-top\\">
        <div class=\\"shadow-sm w-5 h-5 rounded border flex justify-center items-center mt-[3px] border-slate-900\\"><input type=\\"checkbox\\" class=\\"opacity-0 absolute\\"><svg class=\\"w-3 h-3 pointer-events-none stroke-slate-900 stroke-2\\" viewBox=\\"0 0 14 14\\" style=\\"\\">
            <path d=\\"M0.5 8.5498L3.23 12.0598C3.32212 12.1795 3.44016 12.2768 3.57525 12.3443C3.71034 12.4119 3.85898 12.4479 4.01 12.4498C4.15859 12.4515 4.3057 12.4201 4.44063 12.3578C4.57555 12.2956 4.6949 12.204 4.79 12.0898L13.5 1.5498\\" fill=\\"none\\"></path>
          </svg></div>
        <div class=\\"flex w-fit\\"></div>
      </div><label for=\\"Organismes\\">Organismes</label>
    </div>
  </div>
</div>"
`;
